Add note about '+/+' and url address

prko opened the pull request:
<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->

## Purpose and Motivation
While testing #6312 and #6639, I found the following code does not work on Windows due to `+/+`:

```supercollider
(
var url = "https://raw.githubusercontent.com/supercollider/supercollider"
	+/+ "refs/heads/develop/sounds/a11wlk01.wav";
var localPath = "~/Downloads".standardizePath +/+ url.split($/).last;
/// var localPath = Platform.defaultTempDir +/+ url.split($/).last;
var finishedFunc = {
	localPath.openOS;
	"downloaded to %".format(localPath).postln
	};
var errorFunc = {"error".postln };
var progressFunc = { |receivedBytes, totalBytes|
	var downloadProgress = (receivedBytes / totalBytes * 100).round(1e-2);
	"Downloaded %\\%".format(downloadProgress).postln;
};

d = Download(url, localPath, finishedFunc, errorFunc, progressFunc)
)
``` 
I believe there may be other users who, like myself, separate long URL addresses and concatenate the fragments using `+/+`. This pull request aims to prevent such misuses.

<!-- Please provide a description, even if you are linking to a related Issue or PR. -->
<!-- If this fixes an open issue, link to it by writing "Fixes #555." -->

## Types of changes

<!-- Delete lines that don't apply -->

- Documentation

## To-do list

<!-- Complete an item by checking it: [x]. Add new entries to track your progress -->

- [x] Code is tested
- [x] All tests are passing
- [x] Updated documentation
- [x] This PR is ready for review <!-- If not ready for review, consider making a Draft PR instead, or describe what kind of specific feedback you want. -->

<!-- ## Merge notes
(optional) You may also add notes of:
     - Dependencies: if this PR depends on any other PRs or actions
     - Merge Note: any considerations regarding merging this PR
     - Release Notes: text you think would be useful to include in the Release Notes
-->

<!-- Note: The recommended commit message format is:
     component: sub-component: short description of changes
For example:
     classlib: ClassBrowser: fix search with empty query string
Common `comp:` labels:
     help, classlib, sclang, scsynth, plugins, tests, docs, SCDoc
See wiki for more info: Wiki->Creating Pull Requests->git-commit-message-format
-->

telephon commented:
I would guess it doesn't work because of `url.split($/)`, which should be `url.split(Platform.pathSeparator). Can you try this?

prko commented:
The following does not work:
```
(
var url = "https://raw.githubusercontent.com/supercollider/supercollider"
	+/+ "refs/heads/develop/sounds/a11wlk01.wav";
var localPath = "~/Downloads".standardizePath +/+ url.split(Platform.pathSeparator).last;
/// var localPath = Platform.defaultTempDir +/+ url.split($/).last;
var finishedFunc = {
	localPath.openOS;
	"downloaded to %".format(localPath).postln
	};
var errorFunc = {"error".postln };
var progressFunc = { |receivedBytes, totalBytes|
	var downloadProgress = (receivedBytes / totalBytes * 100).round(1e-2);
	"Downloaded %\\%".format(downloadProgress).postln;
};

d = Download(url, localPath, finishedFunc, errorFunc, progressFunc)
)
```

The following two codes blocks work:
```
(
var url = "https://raw.githubusercontent.com/supercollider/" ++ "supercollider/refs/heads/develop/sounds/a11wlk01.wav";
var localPath = "~/Downloads".standardizePath +/+ url.split($/).last;
/// var localPath = Platform.defaultTempDir +/+ url.split($/).last;
var finishedFunc = {
	localPath.openOS;
	"downloaded to %".format(localPath).postln
	};
var errorFunc = {"error".postln };
var progressFunc = { |receivedBytes, totalBytes|
	var downloadProgress = (receivedBytes / totalBytes * 100).round(1e-2);
	"Downloaded %\\%".format(downloadProgress).postln;
};

d = Download(url, localPath, finishedFunc, errorFunc, progressFunc)
)
```
```
(
var url = "https://raw.githubusercontent.com/supercollider/supercollider/refs/heads/develop/sounds/a11wlk01.wav";
var localPath = "~/Downloads".standardizePath +/+ url.split($/).last;
/// var localPath = Platform.defaultTempDir +/+ url.split($/).last;
var finishedFunc = {
	localPath.openOS;
	"downloaded to %".format(localPath).postln
	};
var errorFunc = {"error".postln };
var progressFunc = { |receivedBytes, totalBytes|
	var downloadProgress = (receivedBytes / totalBytes * 100).round(1e-2);
	"Downloaded %\\%".format(downloadProgress).postln;
};

d = Download(url, localPath, finishedFunc, errorFunc, progressFunc)
)
```

dyfer commented:
Yes, `+/+` should only be used for filesystem paths. Good idea to warn the users!

I'd suggest wording this more decisively:

`Do not use code::+/+:: in URLs or in other situations where forward slash is expected. code::+/+:: should only be used in filesystem paths, where it will resolve to either forward or backward slash, depending on the operating system's requirements.`

prko commented:
@dyfer 
Thanks done in https://github.com/supercollider/supercollider/pull/6836/commits/eb20414465560010ca9e44d9b56f838bbe091dcb!

dyfer reviewed (commented):
I think this is good.

What do you think, @telephon ?

telephon reviewed (approved):
good, of course!
